protected/humhub/helpers/DataTypeHelper.php

```diff
@@ -0,0 +1,628 @@
+    public const CLASS_CHECK_INVALID_TYPE_PARAMETER = 2;
+    public const CLASS_CHECK_VALUE_IS_EMPTY = 4;
+    public const CLASS_CHECK_INVALID_CLASSNAME_PARAMETER = 1;
+    public const CLASS_CHECK_INVALID_TYPE = 8;
+    public const CLASS_CHECK_NON_EXISTING_CLASS = 16;
+    public const CLASS_CHECK_VALUE_IS_INSTANCE = 64;
+
+
+    /**
+     * @param mixed $input
+     * @param string|string[] $types
+     * @param bool|null $requireAll
+     * @param string|null $throw = Name of the argument to be used for InvalidArgumentTypeException or Null if no
+     *     exception should be thrown
+     * @param array|null $typesChecked
+     *
+     * @return string|null
+     * @since 1.16
+     */
+    public static function checkType($input, $types, ?bool $requireAll = false, ?string $throw = '$input', ?array &$typesChecked = null): ?string
+    {
+
+        $requireAll ??= false;
+        $returnIndex = false;
+
+        if (is_array($types)) {
+            if (true === ($types[array_key_last($types)] ?? false)) {
+                $returnIndex = array_pop($types);
+            }
+
+            $typesChecked = $types;
+        } else {
+            $typesChecked = self::parseTypes($types);
+        }
+
+        if ($input === null) {
+            if (in_array($i = null, $typesChecked, true) || in_array($i = 'NULL', $typesChecked, true)) {
+                if ($requireAll && count($typesChecked)) {
+                    throw new LogicException("A variable can never be NULL and any other type at the same time!");
+                }
+
+                if ($returnIndex) {
+                    return array_search($i, $typesChecked, true);
+                }
+
+                return 'NULL';
+            }
+        } else {
+            $type = null;
+            $current = gettype($input);
+
+            foreach ($typesChecked as $i => $type) {
+                if ($type === null) {
+                    continue;
+                }
+
+                $type = static::checkTypeHelper($current, $type, $input);
+
+                if ($requireAll) {
+                    if ($type === null) {
+                        break;
+                    }
+                } elseif ($type !== null) {
+                    return $returnIndex
+                        ? $i
+                        : $type;
+                }
+            }
+        }
+
+        $typesChecked = array_map(
+            static fn($item) => is_callable($item, true, $name)
+                ? $name
+                : $item,
+            $typesChecked
+        );
+
+        if ($requireAll && $type !== null) {
+            return implode('|', $typesChecked);
+        }
+
+        if (!$throw) {
+            return null;
+        }
+
+        throw new InvalidArgumentTypeException(
+            $throw,
+            $typesChecked,
+            $input
+        );
+    }
+
+    protected static function checkTypeHelper(string $current, $type, &$input): ?string
```

Comment:
```suggestion
    private static function checkTypeHelper(string $current, $type, &$input): ?string
```

Reply:
@martin-rueegg Related to https://github.com/humhub/humhub/pull/6750#discussion_r1435193156